### PR TITLE
Fix: Updated ugc_post_id parsing logic

### PR DESCRIPTION
**Pull request by bthomson22**
Pull Request
**Are you a current Fivetran customer?** 

Brandon Thomson, Analytics Engineer @ dbt Labs

**What change(s) does this PR introduce?** 

This PR updates the `ugc_post_id` parsing logic within the `stg_linkedin_pages__organization_ugc_post` model. This logic previously only accounted for `share` type posts, and as a result downstream joins were breaking.

**Did you update the CHANGELOG?** 

- [x] Yes

**Does this PR introduce a breaking change?**

- [ ] Yes (please provide breaking change details below.)
- [x] No  (please provide an explanation as to how the change is non-breaking below.)

While this PR doesn't introduce a breaking change, it will result in more accurate data in downstream tables, including `linkedin_pages__posts`.

**Did you update the dbt_project.yml files with the version upgrade (please leverage standard semantic versioning)? (In both your main project and integration_tests)** 

- [x] Yes

**Is this PR in response to a previously created Bug or Feature Request**

- [x] Yes, Issue/Feature https://github.com/fivetran/dbt_linkedin_pages_source/issues/6
- [ ] No 

**How did you test the PR changes?** 

 

- [ ] CircleCi  
- [x] Local (please provide additional testing details below)

Tested changes in my local repo. Test outputs below:
image

**Select which warehouse(s) were used to test the PR**
 

- [ ] BigQuery
- [ ] Redshift
- [x] Snowflake
- [ ] Postgres
- [ ] Databricks
- [ ] Other (provide details below)

**Provide an emoji that best describes your current mood**
 
🤠 

**Feedback**

We are so excited you decided to contribute to the Fivetran community dbt package! We continue to work to improve the packages and would greatly appreciate your [feedback](https://www.surveymonkey.com/r/DQ7K7WW) on our existing dbt packages or what you'd like to see next.

### CODE REVIEW

File: models/stg_linkedin_pages__organization_ugc_post.sql [hunk 1]
```diff
@@ -29,7 +29,7 @@ final as (
     select 
         _fivetran_synced,
         organization_id,
-        replace(ugc_post_id, 'urn:li:share:', '') as ugc_post_id,
+        split_part(ugc_post_id, ':', -1) as ugc_post_id,
```
Comment (fivetran-sheringuyen) [inline models/stg_linkedin_pages__organization_ugc_post.sql]: This definitely works for Snowflake! But before merging we will need to make sure the logic will work across different databases. Unfortunately, the current dbt utils split_part function doesn't quite work across all the DBs we test against and will need a PR filed.

Could you try the similar logic you mentioned in your [issue](https://github.com/fivetran/dbt_linkedin_pages_source/issues/6)? Specifically:

```
        case when lower(ugc_post_id) like '%urn:li:share:%' 
                then replace(ugc_post_id, 'urn:li:share:', '')
            when lower(ugc_post_id) like '%urn:li:ugcpost:%'
                then replace(lower(ugc_post_id), 'urn:li:ugcpost:', '')
            else ugc_post_id end as ugc_post_id,
```

**Review by fivetran-sheringuyen — COMMENTED** [review]
Hi [bthomson22](https://github.com/bthomson22), thank you so much for bringing this to our attention! I made a comment in your PR, let me know if the change works for you.

File: models/stg_linkedin_pages__organization_ugc_post.sql (hunk 1 above)
Comment (bthomson22) [reply models/stg_linkedin_pages__organization_ugc_post.sql]: Updated to match the above logic!

Comment (fivetran-sheringuyen) [reply models/stg_linkedin_pages__organization_ugc_post.sql]: Wonderful! Just to double confirm, does this logic works for you in your DB?

Comment (bthomson22) [reply models/stg_linkedin_pages__organization_ugc_post.sql]: Yup! Tested in my dev env:

Screen Shot 2022-03-17 at 2 05 15 PM

Screen Shot 2022-03-17 at 2 07 04 PM

**Pull request merged**